src/data/finsupp/basic.lean

```diff
@@ -595,12 +595,20 @@ lemma prod_ite_eq (f : α →₀ M) (a : α) (b : α → M → N) :
   f.prod (λ x v, ite (a = x) (b x v) 1) = ite (a ∈ f.support) (b a (f a)) 1 :=
 by { dsimp [finsupp.prod], rw f.support.prod_ite_eq, }
 
+@[simp] lemma sum_ite_self_eq {N : Type*} [add_comm_monoid N] (f : α →₀ N) (a : α) :
```

Comment:
Could you add an assumption `[decidable_eq α]` to the four ite lemmas around this one? In this file, you have classical open, but one might need to apply this lemma to other situations where the decidable_eq would come from a different instance, and this could create trouble.